tests/extract_demand_test.py

```diff
@@ -0,0 +1,137 @@
+"""
+Test script for extract_production.py
```

Comment:
Could you add these tests to a class to sort them by what they are testing? honestly a preference but makes the tests more readable